app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,28 @@
+class API::V1::UsersController < ApplicationController
+  before_action :set_user, only: %i[show]
+
+  def index
+    @users = User.all
+
+    if @users.empty?
+      render json: { message: 'No users found' }, status: :not_found
+    else
+      render json: @users
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @posts = @user.posts
+
+    if @posts.empty?
+      render json: { message: 'No posts found' }, status: :not_found
+    else
+      render json: @posts
+    end
+  end
+
+  def set_user
+    @user = User.find(params[:id])
```

Comment:
- [ ] Kindly make sure your App has an API endpoint for authentication, for both `registration` and `login` as required for the project.

Reply:
@tresorsawasawa  Thank you very much for your kind review and feedback. Please can you give me some suggestions on how to add endpoints for these two?

Reply:
Answered on Slack

Reply:
@tresorsawasawa  Thank you very much for your answer, I found that tutorial very useful. 👍😊✅